backend/wmg/data/cube_pipeline.py

```diff
@@ -0,0 +1,40 @@
+import sys
+from typing import List
+
+from backend.wmg.data.extract import copy_datasets_to_instance
+from backend.wmg.data.load import update_s3_resources
+from backend.wmg.data.transform import get_cells_by_tissue_type, generate_cell_ordering
+from tests.unit.backend.wmg.fixtures.cube import create_cube
+
+
+def load(dataset_directory: List, group_name: str, validate: bool):
+    """
+    Given the path to a directory containing one or more h5ad files and a group name, call the h5ad loading function
+    on all files, loading/concatenating the datasets together under the group name
+    """
+    pass
+
+
+def load_data_and_create_cube(path_to_datasets: str, corpus_name: str, log_level):
+    """
+    Function to copy H5AD datasets (from a preconfiugred s3 bucket) to the path given then,
+    open, transform, normalize and concatenate them together as a tiledb object with a global gene index
+    under the given corpus name.
+    A cube of summary statistics across genes (but queryable by the given dimensions/data) is then generated
+    and stored under big-cube.
+    ## TODO add function to get cell count totals
+    A per tissue mapping of cell ontologies is generated and the files are copied to s3 under a shared timestamp,.
+    On success the least recent set of files are removed from s3
+    """
+    copy_datasets_to_instance(path_to_datasets)
+    load(path_to_datasets, corpus_name)
+    create_cube(corpus_name)
+    cell_type_by_tissue = get_cells_by_tissue_type(corpus_name)
+    generate_cell_ordering(cell_type_by_tissue)
+    update_s3_resources()
+    return True
```

Comment:
If there is a failure, will the pipeline cleanup intermediate, unfinished output?

Reply:
It doesnt do that currently, since we are spinning up instances to run the pipeline and then terminiating when done it didnt seem necessary to add additional cleanup. Let me know if you think its worth adding clean up and I'll create a ticket